app/controllers/spotlight/concerns/application_controller.rb

```diff
@@ -13,7 +13,11 @@ module ApplicationController
       helper Spotlight::ApplicationHelper
 
       rescue_from CanCan::AccessDenied do |exception|
-        redirect_to main_app.root_url, :alert => exception.message
+        if current_exhibit and !can? :read, current_exhibit
+          redirect_to main_app.new_user_session_url, :alert => exception.message
```

Comment:
I think this will be problematic for logged in users who are not admins/curators of the current group.  It will try to log them in, and then once it realizes they're already logged in it will try to redirect them back to the place that raised the `AccessDenied` error.